readthedocs/search/utils.py

```diff
@@ -168,3 +169,23 @@ def _get_sorted_results(results, source_key='_source'):
     ]
 
     return sorted_results
+
+
+def _get_last_31_days_iter():
+    """Returns iterator for last 31 days (including today)."""
+    today = timezone.now().date()
+    last_30th_day = timezone.now().date() - timezone.timedelta(days=30)
+
+    # this includes the current day also
+    last_31_days_iter = [last_30th_day + timezone.timedelta(days=n) for n in range(31)]
+    return last_31_days_iter
+
+
+def _get_last_31_days_str(date_format):
```

Comment:
We should probably have a default string here, which is the one used everywhere currently.